identity-admin/webapp/components/AccountActions/DropdownMenu.tsx

```diff
@@ -1,16 +1,40 @@
-import React, { useState } from 'react';
+import React, { MutableRefObject, useEffect, useRef, useState } from 'react';
 import { Chevron } from '../Icons/Chevron';
 import { Container, Menu, MenuButton } from './DropdownMenu.style';
 
-export const DropdownMenu: React.FC<{ children: React.ReactNode }> = ({
-  children,
-}) => {
+type DropdownMenuProps = {
+  deleteModalRef: () => MutableRefObject<HTMLDivElement | null>;
+};
+
+export const DropdownMenu: React.FC<{
+  children: React.ReactNode;
+  props: DropdownMenuProps;
+}> = ({ children, props }) => {
   const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
 
   const toggleOpen = () => setIsOpen(wasOpen => !wasOpen);
+  const handleClickOutside = (event: MouseEvent) => {
+    const target = event.target;
+    if (
+      isOpen &&
+      target instanceof Node &&
+      !dropdownRef.current?.contains(target) &&
+      !props.deleteModalRef().current?.contains(target)
```

Comment:
Without it I'm getting this error:

```
TS2345: Argument of type 'EventTarget | null' is not assignable to parameter of type 'Node | null'.   Type 'EventTarget' is missing the following properties from type 'Node': baseURI, childNodes, firstChild, isConnected, and 44 more.
```

In practice it'll always be a `Node` because it's the click event handler, but `MouseEvent` could also be a mouse wheel event which maybe doesn't have a `Node` for its target

Reply:
Would a cast be appropriate?

```ts
  const handleClickOutside = (event: MouseEvent) => {
    const target = event.target as Node;
    if (
      isOpen &&
      !dropdownRef.current?.contains(target) &&
      !deleteModalRef.current?.contains(target)
    ) {
      setIsOpen(false);
    }
  };
```